packages/plugin-essentials/sources/index.ts

```diff
@@ -65,6 +65,39 @@ export interface Hooks {
   ) => Promise<void>,
 }
 
+/** @deprecated use Hooks from @yarnpkg/core instead */
+export type Hooks = EssentialHooks;
+
+declare module '@yarnpkg/core' {
+  interface Hooks {
+    afterWorkspaceDependencyAddition?: (
+      workspace: Workspace,
+      target: `${suggestUtils.Target}`,
```

Comment:
@arcanis There's finally a semi-clean solution to get a type out of an enum, solving the problem with multiple enums being instantiated due to virtual packages.
I'm still not happy that this needs to be reduced from an enum to a string union type, but it's already a lot cleaner than what I had to do in the configuration PR with the explicitly written out type union.